File: app/src/main/java/com/chooloo/www/callmanager/ui/settings_fragment/SettingsMvpView.java
package com.chooloo.www.callmanager.ui.settings_fragment;

import android.widget.Toast;

import androidx.preference.Preference;

import com.chooloo.www.callmanager.ui.base.MvpView;

public interface SettingsMvpView extends MvpView {
    void setUp();

    void setListPreferenceSummary(Preference preference, Object newValue);

    void setCheckBoxPreferenceSummary(Preference preferenece, Object newValue);

    void setSwitchPreferenceSummary(Preference preference, Object newValue);

    void setupSimSelection();

    void goToMainActivity();
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/settings_fragment/SettingsMvpView.kt
package com.chooloo.www.callmanager.ui.settings_fragment

import androidx.preference.Preference
import com.chooloo.www.callmanager.ui.base.MvpView

interface SettingsMvpView : MvpView {

    override fun onSetup()

    fun setListPreferenceSummary(preference: Preference, newValue: Any)
    fun setCheckBoxPreferenceSummary(preferenece: Preference, newValue: Any)
    fun setSwitchPreferenceSummary(preference: Preference, newValue: Any)

    fun setupSimSelection()
    fun goToMainActivity()
}